projects/vdk-control-cli/src/vdk/internal/control/command_groups/login_group/login.py

```diff
@@ -134,8 +143,16 @@ def login(
                 countermeasure="Please login providing correct API Token. ",
             )
         else:
-            apikey_auth = ApiKeyAuthentication(api_token_authorization_url, api_token)
-            apikey_auth.authentication_process()
+            apikey_auth = Authentication(
+                authorization_url=api_token_authorization_url,
+                token=api_token,
+                auth_type=auth_type,
+                cache_locally=True,
+            )
+            try:
+                apikey_auth.authenticate()
+            except VDKAuthException as e:
+                click.echo(f"Login failed. Error was: {e.message}")
```

Comment:
That does not seem correct. If auth fails we should exit with non 0 exit code. This will only print the failure and will exit with 0 . Why not let the exception propagate or wrap it in [UserError](https://click.palletsprojects.com/en/7.x/api/#click.UsageError)

Reply:
I agree, though I believe we didn't exit with non-0 status before either.

Reply:
I opted out of propagating the exception as that would be displayed in the user's terminal, and I was not sure if this would be good UX. Also, we have never showed exception messages in terminal for failed login, so doing it now seems like a regression.